api/app/Console/Commands/SendNotificationsAdHocEmail.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Enums\NotificationFamily;
+use App\Models\User;
+use App\Notifications\AdHocEmail;
+use Illuminate\Console\Command;
+use Illuminate\Contracts\Console\PromptsForMissingInput;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+
+class SendNotificationsAdHocEmail extends Command implements PromptsForMissingInput
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'send-notifications:ad-hoc-email
+                                {templateIdEn : The template ID for the English email message}
+                                {templateIdFr : The template ID for the French email message}
+                                {--emailAddress=* : The list of contact email addresses to send the notification to}
+                                {--notificationFamily=* : The list of notification families to send the notification to}
+                                {--notifyAllUsers : Send the notification to all users}
+    ';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Send ad hoc email notifications from given GC Notify templates';
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $successCount = 0;
+        $failureCount = 0;
+
+        $users = $this->getUsersToSendNotificationTo();
+        $userCount = $users->count();
+
+        if ($this->confirm('Do you wish to send notifications to '.$userCount.' users?')) {
+            $progressBar = $this->output->createProgressBar($userCount);
```

Comment:
oo.. progressbar!!!

Reply:
Trying out some of the new Laravel 11 features.  :wink: